weather_mv/loader_pipeline/bq.py

```diff
@@ -154,6 +158,20 @@ def __post_init__(self):
         with open_dataset(self.first_uri, self.xarray_open_dataset_kwargs,
                           self.disable_grib_schema_normalization, self.tif_metadata_for_datetime,
                           is_zarr=self.zarr) as open_ds:
+
+            # Find the grid_resolution. In case of single point we can't find grid resolution.
+            if open_ds['latitude'].size > 1 and open_ds['longitude'].size > 1:
+                # consider that Grid is regular.
+                latitude_length = len(open_ds['latitude'])
+                longitude_length = len(open_ds['longitude'])
+                self.lat_grid_resolution = abs((open_ds["latitude"][-1].values - open_ds["latitude"][0].values
+                                                )/latitude_length)/2
+                self.lon_grid_resolution = abs((open_ds["longitude"][-1].values - open_ds["longitude"][0].values
```

Comment:
I think aI think there may be a succinct way to do this in numpy. For example, consider using this method: 

https://numpy.org/doc/stable/reference/generated/numpy.ptp.html

Reply:
Code updated as per your suggestion. Thanks for the input!